plonky2x/src/frontend/builder/mux.rs

```diff
@@ -0,0 +1,71 @@
+use plonky2::field::extension::Extendable;
+use plonky2::hash::hash_types::RichField;
+
+use crate::prelude::{CircuitBuilder, CircuitVariable, Variable};
+
+trait MuxBuilder {
+    fn select_index<T: CircuitVariable>(&mut self, selector: Variable, inputs: &[T]) -> T;
+}
+
+impl<F: RichField + Extendable<D>, const D: usize> MuxBuilder for CircuitBuilder<F, D> {
+    fn select_index<T: CircuitVariable>(&mut self, selector: Variable, inputs: &[T]) -> T {
+        let num_var_in_t = inputs[0].variables().len();
+        let input_len = inputs.len();
+
+        let mut res = inputs[0].variables();
+        let api = &mut self.api;
+
+        for i in 0..input_len {
+            let target_i = api.constant(F::from_canonical_usize(i));
```

Comment:
nit: here, we always prefer to use the `builder` instead of `api` where a builder method exists. Thus, here I would do `self.constant::<Variable>(F::from_canonical_usize(i))`.

Same with below with `builder.is_equal` and pass in 2 variables (there should be a method on the builder that allows you to compare `equal` on 2 CircuitVariables).

Where the builder method doesn't exist (like `api.select`), then feel free to use the api directly.

Reply:
actually what's the purpose of this thin wrapper, since it leads to some unnecessary oh, e.g. unwrap and wrap, but i adjust to follow the convention.